### PR TITLE
Move static VRF variables to namespace 

### LINKED ISSUE #7729: Refactor VRF static timing variables
Issue overview
--------------
The VRF models have several troublesome static local variables in `InitVRF`,  `CalcVRFCondenser` and `CalcVRFCondenser_FluidTCtrl` which can cause problems with serial unit tests.  Seems they should either be part of the component structs or should be module level variables. Also, at first glance, if there's more than one VRF condenser these variables will be wrong, because `CurrentEndTimeLast` will get updated for the first component and then it will be wrong for the next component? Also, some of these variables are beings stored in ...HRTimer and ...HRTime.

Suggest refactoring the VRF models to use a different approach to track timing for control decisions that does not rely on local statics.

### Details
Some additional details for this issue (if relevant):
 - This came to light due to a serial unit test failure introduce in #7723 which was fixed with a kludge in #7726.
 - Version of EnergyPlus 9.3.0'ish

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jmythms**
Pull request overview
---------------------
 - Partially fixes #7729 .
 - This moves the static variables in the HVACVariableRefrigerantFlow.cc file to the namespace. Multiple condensers might still not work. The HRTime and HRTimer variables are still in the code. 
- This was created because CI was ignoring PR 8339, and created as a clone of PR 8339 without pound sign in the branch name. 

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by jmythms** [comment]
CurrentEndTime and CurrentEndTimeLast for InitVRF and CalcVRFCondenser were extracted while simulating the "VariableRefrigerantFlow_5Zone.idf”. The variables were extracted from [here](https://github.com/NREL/EnergyPlus/blob/f37f9762f1d98546443addfdbd0afd539b5ac8b9/src/EnergyPlus/HVACVariableRefrigerantFlow.cc#L6257) and [here ](https://github.com/NREL/EnergyPlus/blob/f37f9762f1d98546443addfdbd0afd539b5ac8b9/src/EnergyPlus/HVACVariableRefrigerantFlow.cc#L1149).

It was found that the CurrentEndTime variable in [InitVRF ](https://github.com/NREL/EnergyPlus/blob/f37f9762f1d98546443addfdbd0afd539b5ac8b9/src/EnergyPlus/HVACVariableRefrigerantFlow.cc#L6239)and [CalcVRFCondeser ](https://github.com/NREL/EnergyPlus/blob/f37f9762f1d98546443addfdbd0afd539b5ac8b9/src/EnergyPlus/HVACVariableRefrigerantFlow.cc#L1123)were not syncing. The number of calls of CurrentEndTime calls for InitVRF was also found to be more than CalcVRFCondenser.

**Comment by jmythms** [comment]
**Reason**:

[CurrentEndTime in CalcVRFCondenser](https://github.com/NREL/EnergyPlus/blob/f37f9762f1d98546443addfdbd0afd539b5ac8b9/src/EnergyPlus/HVACVariableRefrigerantFlow.cc#L1123) and [CurrentEndTime in InitVRF](https://github.com/NREL/EnergyPlus/blob/f37f9762f1d98546443addfdbd0afd539b5ac8b9/src/EnergyPlus/HVACVariableRefrigerantFlow.cc#L6239) calculates two different values. Therefore, they will not sync. More details about the two variables below. We could just use different names for CurrentEndTime in InitVRF and CalcVRFCondenser to help decrease confusion.

Edit: Reason for mismatch in number of records was that a lot of the calls in CalcVRFCondenser get filtered out when passing through the ["if (!DoingSizing && !WarmupFlag)" line](https://github.com/NREL/EnergyPlus/blob/f37f9762f1d98546443addfdbd0afd539b5ac8b9/src/EnergyPlus/HVACVariableRefrigerantFlow.cc#L1003). 

**Question:**

Can one be substituted for another so that there will be less variables in the namespace?

The CurrentEndTime for CalcVRFCondenser could possibly be also used for InitVRF as well since InitVRF just needs to know if time has gone forward. CurrentEndTime for InitVRF cannot be used for CalcVRFCondenser.
 CurrentEndTime could also be (maybe?) moved to global if this calculation is used elsewhere to calculate time in fractional hours from beginning of simulation till the current system timestep.

**More details about the variables:**

[CurrentEndTIme for InitVRF](https://github.com/NREL/EnergyPlus/blob/f37f9762f1d98546443addfdbd0afd539b5ac8b9/src/EnergyPlus/HVACVariableRefrigerantFlow.cc#L6239) is calculated as

CurrentEndTIme = CurrentTime + DataHVACGlobals::SysTimeElapsed;

Where,

- CurrentTime is the CurrentTime, in fractional hours, from start of day. Uses Loads time step.
- DataHVACGlobals::SysTimeElapsed is the elapsed system time in zone timestep (hours)

InitVRF.CurrentEndTIme is used in an if statement to compare against CurrentEndTimeLast, for finding out if max capacity reset is required. No absolute value for time is needed here, the CurrentEndTime just represents time from start of the day till the current timestep in the simulation and is only used for comparison.

[CurrentEndTime  for CalcVRFCondenser](https://github.com/NREL/EnergyPlus/blob/f37f9762f1d98546443addfdbd0afd539b5ac8b9/src/EnergyPlus/HVACVariableRefrigerantFlow.cc#L1123) is calculated as

CurrentEndTime = double((state.dataGlobal->DayOfSim - 1) * 24) + CurrentTime - TimeStepZone + DataHVACGlobals::SysTimeElapsed;

Where,

- double((state.dataGlobal->DayOfSim - 1) * 24) calculates the number of days from start of simulation to hours
- TimeStepZone is the Zone time step in fractional hours
- CurrentTime is the CurrentTime, in fractional hours, from start of day. Uses Loads time step
- DataHVACGlobals::SysTimeElapsed is the elapsed system time in zone timestep (hours)

CalcVRFCondenser.CurrentEndTime is used as an value of time that has passed from the start of the simulation to calculate the SUMultiplier variable which eventually helps in heat recovery calculations.

File: src/EnergyPlus/HVACVariableRefrigerantFlow.cc [hunk 1]
```diff
@@ -199,6 +199,7 @@ namespace HVACVariableRefrigerantFlow {
     Real64 LoopDXCoolCoilRTF(0.0);          // holds value of DX cooling coil RTF
     Real64 LoopDXHeatCoilRTF(0.0);          // holds value of DX heating coil RTF
     Real64 CondenserWaterMassFlowRate(0.0); // VRF water-cooled condenser mass flow rate (kg/s)
+    Real64 CurrentEndTimeLast;              // end time of last time step
```
Comment (mitchute) [inline src/EnergyPlus/HVACVariableRefrigerantFlow.cc:202]: Good. Hoisting these up to the namespace level is the right path forward here.

File: src/EnergyPlus/HVACVariableRefrigerantFlow.cc [hunk 2]
```diff
@@ -1119,7 +1118,7 @@ namespace HVACVariableRefrigerantFlow {
                 VRF(VRFCond).HRTimer = 0.0;
             }
 
-            // calculate end time of current time step to determine if max capacity reset is required
+            // Calculate the capacity modification factor (SUMultiplier) for the HR mode transition period
```
Comment (mitchute) [inline src/EnergyPlus/HVACVariableRefrigerantFlow.cc:1121]: Is this comment change still valid?

File: src/EnergyPlus/HVACVariableRefrigerantFlow.cc [hunk 3]
```diff
@@ -14550,12 +14542,11 @@ namespace HVACVariableRefrigerantFlow {
         LoopDXCoolCoilRTF = 0.0;
         LoopDXHeatCoilRTF = 0.0;
         CondenserWaterMassFlowRate = 0.0;
-
+        CurrentEndTimeLast = 0.0;
```
Comment (mitchute) [inline src/EnergyPlus/HVACVariableRefrigerantFlow.cc:14545]: Nice. More members for the clear_state.

**Review by mitchute — COMMENTED** [review]
@jmythms thanks! This looks ready. I'll run some tests locally then get it merged.

File: src/EnergyPlus/HVACVariableRefrigerantFlow.cc (hunk 2 above)
Comment (jmythms) [reply src/EnergyPlus/HVACVariableRefrigerantFlow.cc:1121]: Yes, the old comment was misleading.

File: src/EnergyPlus/HVACVariableRefrigerantFlow.cc (hunk 3 above)
Comment (jmythms) [reply src/EnergyPlus/HVACVariableRefrigerantFlow.cc:14545]: Is this bad? 😅

Comment (mitchute) [reply src/EnergyPlus/HVACVariableRefrigerantFlow.cc:14545]: Nope, it's all good :)

Comment (jmythms) [reply src/EnergyPlus/HVACVariableRefrigerantFlow.cc:14545]: Ok 😁

**Comment by mitchute** [comment]
Local tests look good. Merging.

**Pull request merged**